### DISCUSSION TITLE
How to know if my CPU supports BLAS?

### DISCUSSION THREAD

**Discussion by AayushSameerShah** [abetlen/llama-cpp-python#805] [Q&A]
I am trying to run the project on Linux and I installed the library using following:
```python
CMAKE_ARGS="-DLLAMA_BLAS=ON -DLLAMA_BLAS_VENDOR=OpenBLAS" pip install llama-cpp-python
```
But when I run load the model it shows:

> *AVX = 1 | AVX2 = 1 | AVX512 = 0 | AVX512_VBMI = 0 | AVX512_VNNI = 0 | FMA = 1 | NEON = 0 | ARM_FMA = 0 | F16C = 1 | FP16_VA = 0 | WASM_SIMD = 0 | **`BLAS = 0`** | SSE3 = 1 | SSSE3 = 1 | VSX = 0 |*

***Which clearly states BLAS=0 and that is not activated***.

## Q: How to know if my CPU supports BLAS?
Is there any way to know beforehand? I got no error, nothing.

### Instance information
```sh
Architecture:        x86_64
CPU op-mode(s):      32-bit, 64-bit
Byte Order:          Little Endian
CPU(s):              4
On-line CPU(s) list: 0-3
Thread(s) per core:  2
Core(s) per socket:  2
Socket(s):           1
NUMA node(s):        1
Vendor ID:           GenuineIntel
CPU family:          6
Model:               85
Model name:          Intel(R) Xeon(R) Platinum 8259CL CPU @ 2.50GHz
Stepping:            7
CPU MHz:             3105.462
BogoMIPS:            4999.98
Hypervisor vendor:   KVM
Virtualization type: full
L1d cache:           32K
L1i cache:           32K
L2 cache:            1024K
L3 cache:            36608K
NUMA node0 CPU(s):   0-3
Flags:               fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ss ht syscall nx pdpe1gb rdtscp lm constant_tsc rep_good nopl xtopology nonstop_tsc cpuid aperfmperf tsc_known_freq pni pclmulqdq ssse3 fma cx16 pcid sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand hypervisor lahf_lm abm 3dnowprefetch invpcid_single pti fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid mpx avx512f avx512dq rdseed adx smap clflushopt clwb avx512cd avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves ida arat pku ospke
```

### One more note...
When I monitor the CPU usage while inference, it **only uses 50%** of the CPU. 

Please help me get out of this.
Thanks!

**Comment by jurys22** [comment]
I think you want to use BLAS for GPU acceleration https://developer.nvidia.com/cublas

**Reply by AayushSameerShah** [reply]
No luck for CPU 🤔 ? OpenBLAS or CloseBLAS or any blas!!? 😅

**Reply by jurys22** [reply]
The openblas shoud normally work with CPU.  https://github.com/OpenMathLib/OpenBLAS
